### DISCUSSION TITLE
Error occurred while handling an API request in lazer tournament client

### DISCUSSION THREAD

**Discussion by myaimRobotSkin** [ppy/osu#15373] [Q&A] [answered]
I have a problem. When i am trying to add any maps to map pool, i am having this error: "Error occurred while handling an API request"
How do i fix this?
[https://www.youtube.com/watch?v=f1SRak_tKnA](url)

**Comment by peppy** [comment, marked as answer]
Could you please provide logs? See https://github.com/ppy/osu/discussions/15188 for more information on where to find them.

**Reply by myaimRobotSkin** [reply]
[runtime.log](https://github.com/ppy/osu/files/7446256/runtime.log)
[network.log](https://github.com/ppy/osu/files/7446261/network.log)
[performance.log](https://github.com/ppy/osu/files/7446265/performance.log)

**Reply by bdach** [reply]
From the log:

```
2021-10-30 07:58:00 [error]: System.ArgumentNullException: Value cannot be null. (Parameter 'value')
2021-10-30 07:58:00 [error]: at osu.Framework.IO.Network.WebRequest.AddParameter(String name, String value, RequestParameterType type)
2021-10-30 07:58:00 [error]: at osu.Framework.IO.Network.WebRequest.AddParameter(String name, String value)
2021-10-30 07:58:00 [error]: at osu.Game.Online.API.Requests.GetBeatmapRequest.CreateWebRequest()
2021-10-30 07:58:00 [error]: at osu.Game.Online.API.APIRequest.Perform(IAPIProvider api)
2021-10-30 07:58:00 [error]: at osu.Game.Online.API.APIAccess.handleRequest(APIRequest req)
```

I'm not sure how to make something like this happen, however. @myaimRobotSkin mind providing your `bracket.json` file from the tournament client itself?

**Reply by myaimRobotSkin** [reply]
`{
  "Ruleset": {
    "ID": 0,
    "Name": "osu!",
    "ShortName": "osu",
    "InstantiationInfo": "osu.Game.Rulesets.Osu.OsuRuleset, osu.Game.Rulesets.Osu"
  },
  "Matches": [
    {
      "ID": 1,
      "Team1Acronym": "RAS",
      "Team1Score": 0,
      "Team2Acronym": "MIL",
      "Team2Score": 0,
      "Completed": false,
      "Losers": false,
      "PicksBans": [],
      "Current": true,
      "Date": "2021-10-30T15:00:00+10:00",
      "ConditionalMatches": [],
      "Position": {
        "X": 156,
        "Y": 338
      },
      "Acronyms": [
        "RAS",
        "MIL"
      ],
      "WinnerColour": "Blue",
      "PointsToWin": 5
    }
  ],
  "Rounds": [
    {
      "Name": "Round of 32",
      "Description": "lol",
      "BestOf": 9,
      "Beatmaps": [],
      "StartDate": "2021-10-30T15:00:00+10:00",
      "Matches": [
        1
      ]
    }
  ],
  "Teams": [
    {
      "FullName": "Rasty_game",
      "FlagName": "rasty",
      "Acronym": "RAS",
      "SeedingResults": [],
      "Seed": "1",
      "LastYearPlacing": 1,
      "AverageRank": 717266.0,
      "Players": [
        {
          "id": 15223115,
          "username": "Rasty_game",
          "country": {
            "name": "Russian Federation",
            "code": "RU"
          },
          "Status": null,
          "Activity": null,
          "cover": {
            "url": "https://osu.ppy.sh/images/headers/profile-covers/c4.jpg"
          },
          "cover_url": "https://osu.ppy.sh/images/headers/profile-covers/c4.jpg",
          "statistics": {
            "level": {
              "current": 58,
              "progress": 81
            },
            "global_rank": 717266,
            "country_rank": 62887,
            "RankHistory": {
              "mode": "osu",
              "data": [
                764344,
                754966,
                755651,
                655389,
                655004,
                656028,
                657206,
                658078,
                659325,
                660550,
                661685,
                662826,
                660205,
                661206,
                662315,
                663391,
                664538,
                665686,
                666890,
                667937,
                668574,
                669607,
                670606,
                666341,
                667381,
                668482,
                669515,
                670613,
                671744,
                672841,
                673888,
                674966,
                675962,
                676939,
                677989,
                679085,
                680083,
                681104,
                681927,
                682918,
                683808,
                684768,
                685922,
                686901,
                687828,
                688779,
                689684,
                690696,
                691778,
                692930,
                689588,
                689455,
                690375,
                691288,
                692194,
                692906,
                693958,
                694910,
                695692,
                696555,
                697450,
                698266,
                699143,
                700085,
                700946,
                701738,
                702538,
                703396,
                704208,
                705090,
                706031,
                706894,
                707691,
                708447,
                709240,
                710049,
                710962,
                711880,
                712725,
                713447,
                714300,
                715134,
                715950,
                716818,
                717787,
                718665,
                719432,
                720226,
                716534,
                717266
              ]
            },
            "pp": 767.452,
            "ranked_score": 125350879,
            "hit_accuracy": 90.5087,
            "play_count": 4208,
            "play_time": 237064,
            "total_score": 1339127743,
            "total_hits": 604217,
            "maximum_combo": 959,
            "grade_counts": {
              "ssh": 1,
              "sh": 5,
              "s": 7,
              "a": 31
            }
          }
        }
      ]
    },
    {
      "FullName": "miltipl",
      "FlagName": "milti",
      "Acronym": "MIL",
      "SeedingResults": [],
      "Seed": "2",
      "LastYearPlacing": 1,
      "Players": [
        {
          "id": 18576733,
          "username": "miltipl",
          "country": {
            "name": "Russian Federation",
            "code": "RU"
          },
          "Status": null,
          "Activity": null,
          "cover": {
            "url": "https://osu.ppy.sh/images/headers/profile-covers/c6.jpg",
            "id": 6
          },
          "cover_url": "https://osu.ppy.sh/images/headers/profile-covers/c6.jpg",
          "statistics": {
            "level": {
              "current": 10,
              "progress": 14
            },
            "RankHistory": {
              "mode": "osu",
              "data": [
                3797117,
                3803590,
                3810190,
                3153315,
                3162096,
                3170853,
                3179464,
                3188150,
                3197641,
                3207142,
                3215976,
                3224365,
                3232766,
                3241567,
                3250459,
                3259575,
                3268873,
                3277605,
                3285522,
                3293934,
                3302819,
                3311678,
                3320780,
                3329935,
                3338224,
                3346192,
                3354165,
                3362278,
                3370250,
                3378663,
                3386938,
                3394742,
                3402107,
                3409429,
                3416894,
                3424599,
                3432486,
                3440290,
                3447616,
                3454672,
                3461428,
                3468119,
                3474683,
                3481510,
                3488450,
                3495267,
                3501521,
                3507856,
                3514517,
                3521014,
                3527595,
                3534437,
                3540843,
                3547049,
                3553501,
                3559912,
                3566273,
                3572574,
                3579205,
                3585508,
                3591703,
                3597946,
                3604219,
                3610586,
                3617211,
                3623869,
                3630260,
                3636488,
                3643310,
                3650071,
                3657320,
                3664213,
                3671540,
                3678772,
                3685826,
                3693151,
                3700278,
                3707953,
                3715542,
                3722976,
                3730119,
                3737281,
                3744469,
                3751291,
                3757758,
                3764179,
                3770683,
                3777192,
                3783820,
                3790526
              ]
            },
            "pp": 0.0,
            "ranked_score": 1014684,
            "hit_accuracy": 80.7251,
            "play_count": 105,
            "play_time": 3709,
            "total_score": 6459956,
            "total_hits": 5973,
            "maximum_combo": 205,
            "grade_counts": {
              "ssh": 0,
              "sh": 0
            }
          }
        }
      ]
    }
  ],
  "Progressions": [],
  "ChromaKeyWidth": 1366,
  "PlayersPerTeam": 4
}`

**Reply by bdach** [reply]
Actually, I got confused. This is on the 2020.1028.0 release, wherein [this pull](https://github.com/ppy/osu/pull/15319) wasn't merged yet. It should be working fine after it (i.e. next release)

**Reply by peppy** [reply]
For now the recommendation would be to use the previous release. We'll get a new one out within the next day or two to fix this.
